### ISSUE TITLE
Stack overflow in parser

### ISSUE THREAD

**Issue by artursapek** [babel/babel#1149]
First off, congrats on the 5.0.0 release! We use babel to build Codecademy and have been very happy with it.

However, we just tried upgrading to 5.0.0 and got a stack overflow building our app. I've reduced it to the following:

```es6
export default {
  throwUp: function(file) {
    var file = file[0];
  }
};
```

[Sandbox link](http://babeljs.io/repl/#?experimental=true&evaluate=true&loose=false&spec=false&code=export%20default%20%7B%0A%20%20throwUp%3A%20function(file)%20%7B%0A%20%20%20%20var%20file%20%3D%20file%5B0%5D%3B%0A%20%20%7D%0A%7D%3B%0A)

It appears the re-assignment makes the parser explode. It yields the following:

```
$ babel test.js
/usr/local/lib/node_modules/babel/node_modules/babel-core/lib/babel/helpers/parse.js:70
    throw err;
          ^
RangeError: test.js: Maximum call stack size exceeded
```

I realize the code actually shouldn't have the `var`, so this isn't a huge issue because I can fix it on my side. But I bet this will catch others off guard as well so it's probably worth fixing. This mistake is easy to make.

I may take a crack at fixing this later myself if I have some spare time, but I bet someone on the team would be able to pinpoint the patch a lot faster than me :)

Again, thanks for continuing to build babel.

**Comment by sebmck** [comment]
It's not actually in the parser. Has to do with faulty binding resolution. Should be an easy fix.

**Comment by sebmck** [comment]
I've disabled it for the timebeing. It's a part of a WIP path resolution system. I'll push out a new release shortly.

**Comment by sebmck** [comment]
Fixed as of 5.0.7, thanks!

**Comment by artursapek** [comment]
Awesome, thank you!
